plugins/obs-ffmpeg/obs-ffmpeg-mux.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <obs-avc.h>
+#include <obs-module.h>
+#include <obs-hotkey.h>
+#include <util/circlebuf.h>
+#include <util/darray.h>
+#include <util/dstr.h>
+#include <util/pipe.h>
+#include <util/platform.h>
+#include <util/threading.h>
+
+struct ffmpeg_muxer {
+	obs_output_t *output;
+	os_process_pipe_t *pipe;
+	int64_t stop_ts;
+	uint64_t total_bytes;
+	struct dstr path;
+	bool sent_headers;
+	volatile bool active;
+	volatile bool stopping;
+	volatile bool capturing;
+
+	/* replay buffer */
+	struct circlebuf packets;
+	int64_t cur_size;
+	int64_t cur_time;
+	int64_t max_size;
+	int64_t max_time;
+	int64_t save_ts;
+	int keyframes;
+	obs_hotkey_id hotkey;
+
+	DARRAY(struct encoder_packet) mux_packets;
+	pthread_t mux_thread;
+	bool mux_thread_joinable;
+	volatile bool muxing;
+
+	/* HLS */
+	struct dstr printable_path;
+	struct dstr stream_key;
+	struct dstr muxer_settings;
```

Comment:
Since printable_path, stream_key and muxer_settings are also used in the obs-ffmpeg-mux.c file, now I think it's better to keep them with first block fields indicating that they are shared.